keps/sig-storage/4876-mutable-csinode-allocatable/kep.yaml

```diff
@@ -0,0 +1,50 @@
+title: Mutable CSINode Allocatable Property
+kep-number: 4876
+authors:
+  - "@torredil"
+owning-sig: sig-storage
+participating-sigs:
```

Comment:
I think cluster-autoscaler might be partially broken with this proposal, as it might be using the counters to predict how many slots there will be on a new node.

cc @MaciekPytel @x13n, could you confirm?

Reply:
Cluster Autoscaler doesn't directly use CSINode objects at all. During scale up simulations, only Node object is created in-memory by CA and then kube-scheduler code is executed on it to make a scale up decision. If I understand this proposal correctly, no changes to kube-scheduler or its plugins are a part of this proposal, so I'd expect CA to continue working. Since validating scheduler keeps working is a beta requirement of this KEP, I'd suggest adding a similar validation of CA behavior.

Reply:
Thank you for the insight @x13n -- that is correct and lines up with my understanding as well, no changes to kube-scheduler or its plugins are needed / part of this proposal.

> I'd suggest adding a similar validation of CA behavior.

Agree, added CA validation to the beta requirements.